File: app/src/main/java/com/sakurafish/pockettoushituryou/shared/Pref.java
package com.sakurafish.pockettoushituryou.shared;

import android.content.SharedPreferences;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public class Pref {
    private SharedPreferences preferences;

    @Inject
    public Pref(SharedPreferences preferences) {
        this.preferences = preferences;
    }

    public void setPref(final String key, final String value) {
        final SharedPreferences.Editor editor = getEditor();
        getEditor().putString(key, value);
        editor.commit();
    }

    public String getPrefString(final String key) {
        return this.preferences.getString(key, null);
    }

    public void setPref(final String key, boolean value) {
        final SharedPreferences.Editor editor = getEditor();
        editor.putBoolean(key, value);
        editor.commit();
    }

    public boolean getPrefBool(final String key, final boolean defaultBool) {
        return this.preferences.getBoolean(key, defaultBool);
    }

    public void setPref(final String key, final int value) {
        final SharedPreferences.Editor editor = getEditor();
        editor.putInt(key, value);
        editor.commit();
    }

    public int getPrefInt(final String key) {
        return this.preferences.getInt(key, 0);
    }

    public long getPrefLong(final String key, final long defaultValue) {
        return this.preferences.getLong(key, defaultValue);
    }

    public void setPrefLong(final String key, final long value) {
        final SharedPreferences.Editor editor = getEditor();
        editor.putLong(key, value);
        editor.commit();
    }

    public SharedPreferences.Editor getEditor() {
        return this.preferences.edit();
    }
}

File: app/src/main/java/com/sakurafish/pockettoushituryou/shared/Pref.kt
package com.sakurafish.pockettoushituryou.shared

import android.content.SharedPreferences
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class Pref @Inject constructor(private val preferences: SharedPreferences) {

    fun setPref(key: String?, value: String?) {
        val editor = editor
        editor.putString(key, value)
        editor.commit()
    }

    fun getPrefString(key: String?): String? {
        return preferences.getString(key, null)
    }

    fun setPref(key: String?, value: Boolean) {
        val editor = editor
        editor.putBoolean(key, value)
        editor.commit()
    }

    fun getPrefBool(key: String?, defaultBool: Boolean): Boolean {
        return preferences.getBoolean(key, defaultBool)
    }

    fun setPref(key: String?, value: Int) {
        val editor = editor
        editor.putInt(key, value)
        editor.commit()
    }

    fun getPrefInt(key: String?): Int {
        return preferences.getInt(key, 0)
    }

    fun getPrefLong(key: String?, defaultValue: Long): Long {
        return preferences.getLong(key, defaultValue)
    }

    fun setPrefLong(key: String?, value: Long) {
        val editor = editor
        editor.putLong(key, value)
        editor.commit()
    }

    val editor: SharedPreferences.Editor
        get() = preferences.edit()
}